searx/engines/wttr.py

```diff
@@ -18,118 +20,114 @@
 
 url = "https://wttr.in/{query}?format=j1&lang={lang}"
 
-
-def get_weather_condition_key(lang):
-    if lang == "en":
-        return "weatherDesc"
-
-    return "lang_" + lang.lower()
-
-
-def generate_day_table(day):
-    res = ""
-
-    res += f"<tr><td>{gettext('Average temp.')}</td><td>{day['avgtempC']}°C / {day['avgtempF']}°F</td></tr>"
-    res += f"<tr><td>{gettext('Min temp.')}</td><td>{day['mintempC']}°C / {day['mintempF']}°F</td></tr>"
-    res += f"<tr><td>{gettext('Max temp.')}</td><td>{day['maxtempC']}°C / {day['maxtempF']}°F</td></tr>"
-    res += f"<tr><td>{gettext('UV index')}</td><td>{day['uvIndex']}</td></tr>"
-    res += f"<tr><td>{gettext('Sunrise')}</td><td>{day['astronomy'][0]['sunrise']}</td></tr>"
-    res += f"<tr><td>{gettext('Sunset')}</td><td>{day['astronomy'][0]['sunset']}</td></tr>"
-
-    return res
-
-
-def generate_condition_table(condition, lang, current=False):
-    res = ""
-
-    if current:
-        key = "temp_"
-    else:
-        key = "temp"
-
-    res += (
-        f"<tr><td><b>{gettext('Condition')}</b></td>"
-        f"<td><b>{condition[get_weather_condition_key(lang)][0]['value']}</b></td></tr>"
-    )
-    res += (
-        f"<tr><td><b>{gettext('Temperature')}</b></td>"
-        f"<td><b>{condition[key+'C']}°C / {condition[key+'F']}°F</b></td></tr>"
-    )
-    res += (
-        f"<tr><td>{gettext('Feels like')}</td><td>{condition['FeelsLikeC']}°C / {condition['FeelsLikeF']}°F</td></tr>"
-    )
-    res += (
-        f"<tr><td>{gettext('Wind')}</td><td>{condition['winddir16Point']} — "
-        f"{condition['windspeedKmph']} km/h / {condition['windspeedMiles']} mph</td></tr>"
-    )
-    res += (
-        f"<tr><td>{gettext('Visibility')}</td><td>{condition['visibility']} km / {condition['visibilityMiles']} mi</td>"
-    )
-    res += f"<tr><td>{gettext('Humidity')}</td><td>{condition['humidity']}%</td></tr>"
-
-    return res
+# adapted from https://github.com/chubin/wttr.in/blob/master/lib/constants.py
+WWO_TO_CONDITION = {
+    "113": "clear sky",
+    "116": "partly cloudy",
+    "119": "cloudy",
+    "122": "very cloudy",
+    "143": "fog",
+    "176": "light showers",
+    "179": "light sleet showers",
+    "182": "light sleet",
+    "185": "light sleet",
+    "200": "rain and thunder",
+    "227": "light snow",
+    "230": "heavy snow",
+    "248": "fog",
+    "260": "fog",
+    "263": "light showers",
+    "266": "light rain",
+    "281": "light sleet",
+    "284": "light sleet",
+    "293": "light rain",
+    "296": "light rain",
+    "299": "heavy showers",
+    "302": "heavy rain",
+    "305": "heavy showers",
+    "308": "heavy rain",
+    "311": "light sleet",
+    "314": "light sleet",
+    "317": "light sleet",
```

Comment:
wttr.in has only a small set of symbols, a more detailed mapping is given in ..

- https://www.worldweatheronline.com/feed/wwoConditionCodes.txt
- https://www.worldweatheronline.com/weather-api/api/docs/weather-icons.aspx